smoothie-lifecycle-viewmodel-ktp/build.gradle

```diff
@@ -0,0 +1,47 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+repositories {
+  google()
+  jcenter()
+}
+
+android {
+  compileSdkVersion 28
+
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_7
+    targetCompatibility JavaVersion.VERSION_1_7
+  }
+
+  defaultConfig {
+    minSdkVersion 15
+  }
+
+  sourceSets {
+    main.java.srcDirs += 'src/main/kotlin'
+    test.java.srcDirs += 'src/test/kotlin'
+  }
+
+  lintOptions {
+    textReport true
+    textOutput 'stdout'
+    // We run a full lint analysis as build part in CI, so skip vital checks for assemble tasks.
+    checkReleaseBuilds false
+  }
+}
+
+dependencies {
+  api project(':smoothie-lifecycle-viewmodel')
+  implementation project(':ktp')
+  implementation deps.lifecycle_viewmodel
+  implementation deps.lifecycle_extensions
+  implementation deps.androidxv4
+  implementation deps.kotlin_runtime
```

Comment:
we definitely need a convention for this. api / implementation / compileOnly all make sense. Let's take the best practice from other libs.